test/stubs/StubRuleViewer.ts

```diff
@@ -0,0 +1,19 @@
+import {Rule} from '@salesforce/code-analyzer-core';
+import {RuleViewer} from '../../src/lib/viewers/RuleViewer';
+
+
+export class StubRuleViewer implements RuleViewer {
```

Comment:
Maybe call this SpyRuleViewer... since a spy collects what has been passed in. And then either make the viewedRules public... or add a getter so that the test can access the results.

A trick I do with almost all of my spy classes is to add the input arguments as an object array for each of the methods.

For example:

    public viewCallHistory: {rules: Rule[]}[] = [];

And then

	public view(rules: Rule[]): void {
		this.viewCallHistory.push({rules});
	}


This way you can check how many times the method is called by checking the call history length and then get the values.